pkg/customers/customers.go

```diff
@@ -214,8 +202,16 @@ func (rep *customerRepresentative) validate() error {
 }
 
 func (req customerRequest) validate() error {
-	if req.FirstName == "" || req.LastName == "" {
-		return errors.New("invalid customer fields: empty name field(s)")
+	if req.Type == client.CUSTOMERTYPE_INDIVIDUAL ||
+		(req.Type == client.CUSTOMERTYPE_BUSINESS &&
+			req.BusinessType == client.BUSINESSTYPE_INDIVIDUAL_SOLE_PROPRIETOR_OR_SINGLE_MEMBER_LLC) {
+		if req.FirstName == "" || req.LastName == "" {
+			return errors.New("invalid customer fields: empty name field(s)")
+		}
+	} else {
+		if req.BusinessName == "" {
+			return errors.New("invalid customer fields: empty business name field")
+		}
```

Comment:
The nested-ifs and bool statements are a bit hard to read at first glance. I recommend refactoring this to something like:

```go
	isIndividualOrLLC := req.Type == client.CUSTOMERTYPE_INDIVIDUAL || req.BusinessType = client.BUSINESSTYPE_INDIVIDUAL_SOLE_PROPRIETOR_OR_SINGLE_MEMBER_LLC
	if isIndividualOrLLC && (req.FirstName == "" || req.LastName == ""){
	         return errors.New("empty name field(s)")
	} else if req.BusinessName == "" {
		return errors.New("empty business name")
	}

```